### PR TITLE
[kubeappsapis] Add kubeappsapis in CI

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

~(PR not for the main branch yet, targeting branch: 3085-useNewAPI).~

This PR adds some changes in order to get the tests to work when testing code that depends on the new API (as #3164)

### Benefits

Tests should be passing again...

### Possible drawbacks

... however, these changes have been extracted from #3164 but this PR is not passing yet :(

### Applicable issues

   - related #3165

### Additional information

~I'll keep this PR as a draft until I get my PR working~

### CODE REVIEW

**Review by absoludity — COMMENTED** [review]
Thanks Antonio. So from what I can see, this just updates CI to include the kubeapps-apis image in its testing etc.? ie. it's not yet using it for anything in this PR (only in your base branch where you'll be landing the UI changes).

If that's the case, I see on reason why we shouldn't target our main branch (and once tests are passing), land this on our main branch: it is just ensuring that the image builds and (go) unit tests pass, which would be helpful for us right now while developing, even though we're not (yet) releasing the image.

**Comment by absoludity** [comment]
I would additionally set the plugins to `helm` only for now and `redis.enabled=false` manually for CI (or until we're close to using the fluxv2 plugin, update the values.kubeappsapis.yaml with those options).

From the [logged output of the e2e tests](https://app.circleci.com/pipelines/github/antgamdia/kubeapps/932/workflows/f76d895c-013a-44db-b724-08da1785c714/jobs/6752/steps), it appears it's failing/timing out during:
```
k8s_wait_for_job_completed kubeapps apprepositories.kubeapps.com/repo-name=bitnami
```
which seems unrelated. Retrying.

File: chart/kubeapps/templates/kubeappsapis/deployment.yaml [hunk 1]
```diff
@@ -92,6 +92,7 @@ spec:
             # TODO (gfichtenolt) this as required by the kubeapps apis service (which will
             # longer-term pass something to the plugins so that the plugins won't need to
             # know these details). Currently they're used directly by the flux plugin
+            {{- if .Values.redis.enabled }}
```
Comment (absoludity) [inline chart/kubeapps/templates/kubeappsapis/deployment.yaml:95]: Shouldn't that change already be on master from my earlier PR?

**Review by absoludity — APPROVED** [review]
Excellent. Thanks Antonio. +1 when CI is passing.

File: chart/kubeapps/templates/kubeappsapis/deployment.yaml (hunk 1 above)
Comment (antgamdia) [reply chart/kubeapps/templates/kubeappsapis/deployment.yaml:95]: Yep... perhaps it's an issue when merging (in fact, the `test render` CI step is failing; I'll have a look at it soon)

**Pull request merged**